src/snowcli/cli/nativeapp/manager.py

```diff
@@ -64,30 +116,75 @@ def artifacts(self) -> List[ArtifactMapping]:
     def deploy_root(self) -> Path:
         return Path(self.project_root, self.definition["deploy_root"])
 
+    @cached_property
+    def package_scripts(self) -> List[str]:
+        """
+        Relative paths to package scripts from the project root.
+        """
+        return self.definition.get("package", {}).get("scripts", [])
+
+    @cached_property
+    def stage_fqn(self) -> str:
+        return f'{self.package_name}.{self.definition["source_stage"]}'
+
+    @cached_property
+    def stage_schema(self) -> Optional[str]:
+        return extract_schema(self.stage_fqn)
+
+    @cached_property
+    def warehouse(self) -> Optional[str]:
+        return self.definition.get("application", {}).get("warehouse", None)
+
+    @cached_property
+    def project_identifier(self) -> str:
+        return clean_identifier(self.definition["name"])
+
+    @cached_property
+    def package_name(self) -> str:
+        return self.definition.get("package", {}).get(
+            "name", default_app_package(self.project_identifier)
+        )
+
+    @cached_property
+    def app_name(self) -> str:
+        return self.definition.get("application", {}).get(
+            "name", default_application(self.project_identifier)
+        )
```

Comment:
+1 to the property methods introduced above.

Reply:
As I'm working on snowpark changes I'm thinking if we should introduce `NativeApp`, `Function` or `Procedure` objects that would be DTOs. Then the managers are real managers. This change make manager responsibility non-obvious

Reply:
Sure, open to that. Can we treat it as an enhancement?